File: src/main/java/fr/unreal852/quantum/world/QuantumWorld.java
package fr.unreal852.quantum.world;

import fr.unreal852.quantum.world.config.QuantumWorldConfig;
import net.minecraft.server.world.ServerWorld;
import xyz.nucleoid.fantasy.RuntimeWorldHandle;

public class QuantumWorld
{
    private final RuntimeWorldHandle _runtimeWorldHandle;
    private final QuantumWorldConfig _worldConfig;

    public QuantumWorld(RuntimeWorldHandle runtimeWorldHandle, QuantumWorldConfig worldConfig)
    {
        _runtimeWorldHandle = runtimeWorldHandle;
        _worldConfig = worldConfig;
    }

    public RuntimeWorldHandle getRuntimeWorld()
    {
        return _runtimeWorldHandle;
    }

    public ServerWorld getServerWorld()
    {
        return _runtimeWorldHandle.asWorld();
    }

    public QuantumWorldConfig getWorldConfig()
    {
        return _worldConfig;
    }
}

File: src/main/java/fr/unreal852/quantum/world/QuantumWorld.kt
package fr.unreal852.quantum.world

import net.minecraft.server.world.ServerWorld
import xyz.nucleoid.fantasy.RuntimeWorldHandle

class QuantumWorld(val runtimeWorld: RuntimeWorldHandle, val worldData: QuantumWorldData) {
    val serverWorld: ServerWorld
        get() = runtimeWorld.asWorld()
}